packages/debug/src/browser/debug-configuration-manager.ts

```diff
@@ -254,21 +277,27 @@ export class DebugConfigurationManager {
 
     protected async doOpen(model: DebugConfigurationModel): Promise<EditorWidget> {
         let uri = model.uri;
-        if (!uri) {
+        const configFileUri = await this.getFolderConfigurationUri(model);
+        if (!await this.filesystem.exists(configFileUri.toString())) {
```

Comment:
why previous check was not good? if `uri` is missing, that already means that model does not exist.

Reply:
`DebugConfigurationModel.uri` does point to the the config.
In a multi root workspace it points to the workspace file, e.g., `~/dev/ws/myproject.theia-workspace`. And this is causing problems, because we need config uri that points to the folder config such as `~/dev/hello-world/.theia/launch.json`.